feature_engine/selection/eliminate_feature_by_feature_importance.py

```diff
@@ -0,0 +1,234 @@
+        https://scikit-learn.org/stable/modules/model_evaluation.html
+
+    threshold: float, int, default = 0.01
+        The value that defines if a feature will be kept or removed. Note that for
+        metrics like roc-auc, r2_score and accuracy, the thresholds will be floats
+        between 0 and 1. For metrics like the mean_square_error and the
+        root_mean_square_error the threshold will be a big number.
+        The threshold must be defined by the user.
+
+    cv : int, default=3
+        Desired number of cross-validation fold to be used to fit the estimator.
+
+    Attributes
+    ----------
+
+
+    Methods
+    -------
+
+    fit: finds important features
+
+    transform: removes non-important / non-selected features
+
+    fit_transform: finds and removes non-important features
+
+    """
+
+    def __init__(
+        self,
+        estimator=RandomForestClassifier(),
+        scoring="roc_auc",
+        cv=3,
+        threshold=0.01,
+        variables=None,
+        regression=False
+    ):
+
+        if not isinstance(cv, int) or cv < 1:
+            raise ValueError("cv can only take positive integers bigger than 1")
+
+        if not isinstance(threshold, (int, float)):
+            raise ValueError("threshold can only be integer or float")
+
+        if not isinstance(regression, bool):
+            raise ValueError("regression should be a boolean")
+
+        self.variables = _define_variables(variables)
+        self.estimator = estimator
+        self.scoring = scoring
+        self.threshold = threshold
+        self.cv = cv
+        self.regression = regression
+
+    def fit(self, X, y):
+        """
+
+        Args
+        ----
+
+        X: pandas dataframe of shape = [n_samples, n_features]
+           The input dataframe
+
+        y: array-like of shape (n_samples)
+           Target variable. Required to train the estimator.
+
+
+        Returns
+        -------
+
+        self
+        """
+
+        # check input dataframe
+        X = _is_dataframe(X)
+
+        # reset the index
+        X = X.reset_index(drop=True)
+        y = y.reset_index(drop=True)
+
+        # find numerical variables or check variables entered by user
+        self.variables = _find_numerical_variables(X, self.variables)
+
+        # train model with all features and cross-validation
+        model = cross_validate(
+            self.estimator,
+            X,
+            y,
+            cv=self.cv,
+            scoring=self.scoring,
+        )
+
+        # store initial model performance
+        self.initial_model_performance_ = model["test_score"].mean()
+
+        # get performance metric
+        scorer = get_scorer(self.scoring)
+
+        # choose the approriate attibute to use for the appropriate model
+        if (self.regression):
+            get_feature_importance_method = "coef_"
+        else:
+            get_feature_importance_method = "feature_importances_"
```

Comment:
Also, coef_ or importances_ depend on whether it is a tree or a linear model, not on regression vs classification

> Oh, yes Definitely..

I For the coeff importance part. I have a clear idea about how to implement it now.
I'm thinking of adding a new file in the selection folder called "selection_utils"? where I would put the code for the function that returns the feature importance giving a fitted estimator.
Or should I add the function in the __init__.py file in the selection folder ?
---------------------------------------------------------------------------
Where do you see this file existing?

Reply:
for now I would say, add a file called base_selector.py within the selection folder. The classes and functions there should not be called from the init because they are internal.

Then, as more selectors come up, we see if we need to move around. Thank you!